### PR TITLE
Support Julia 1.11

### PR DESCRIPTION
**Pull request by vchuravy**
- Test against v1.11
- WIP: adapt to 1.11 changes

### CODE REVIEW

**Comment by wsmoses** [comment]
This is definitely required, but the bigger issue is almost certainly adapting to the new pipeline changes.

File: src/compiler/interpreter.jl [hunk 1]
```diff
@@ -19,34 +35,38 @@ struct EnzymeInterpreter <: AbstractInterpreter
...
 end
 
 Core.Compiler.InferenceParams(interp::EnzymeInterpreter) = interp.inf_params
 Core.Compiler.OptimizationParams(interp::EnzymeInterpreter) = interp.opt_params
-Core.Compiler.get_world_counter(interp::EnzymeInterpreter) = interp.world
+get_inference_world(interp::EnzymeInterpreter) = interp.world
 Core.Compiler.get_inference_cache(interp::EnzymeInterpreter) = interp.local_cache
-Core.Compiler.code_cache(interp::EnzymeInterpreter) = WorldView(interp.global_cache, interp.world)
+@static if HAS_INTEGRATED_CACHE
+    Core.Compiler.cache_owner(interp::EnzymeInterpreter) = interp.cache_or_token
```
Comment (vchuravy) [inline src/compiler/interpreter.jl:66]:
```suggestion
    Core.Compiler.cache_owner(interp::EnzymeInterpreter) = interp.token
```

File: src/compiler/interpreter.jl [hunk 2]
```diff
@@ -19,34 +35,38 @@ struct EnzymeInterpreter <: AbstractInterpreter
...
+else
+    Core.Compiler.code_cache(interp::EnzymeInterpreter) = WorldView(interp.cache_or_token, interp.world)
```
Comment (vchuravy) [inline src/compiler/interpreter.jl:68]:
```suggestion
    Core.Compiler.code_cache(interp::EnzymeInterpreter) = WorldView(interp.cache, interp.world)
```

**Comment by wsmoses** [comment]
Merging WIP, will continue in separate PRs

**Pull request merged**